nodes/hsm.js

```diff
@@ -0,0 +1,158 @@
+        node.currentHsm = hsm.hsm; 
+        //Alarm gets cancelled
+      } else if (event.name === 'hsmAlert') {
+        node.currentHsm = event.value;
+      } else if (event.name === 'hsmStatus') {
+        node.currentHsm = event.value;
+      }
+      node.log(`HSM: ${this.currentHsm}`);
+
+      if (node.sendEvent) {
+        const msg = {
+          payload: {
+            name: event.name,
+            value: event.value,
+            displayName: event.displayName,
+            descriptionText: event.descriptionText,
+          },
+          topic: 'hubitat-hsm',
+        };
+        node.send(msg);
+      }
+
+      node.status({ fill: 'blue', shape: node.shape, text: this.currentHsm });
+    }).bind(null, this));
+
+    initializeHsm().catch(() => {});
+    
+    function convertAlarmState(value) {
+      switch (value) {
+        case 'stay':
+        case 'armHome':
+        case 'armedHome':
+        case 'armhome':
+        case 'armedhome':
+        case 0:
+            return 'armHome';
+        case 'away':
+        case 'armaway':
+        case 'armAway':
+        case 'armedaway':
+        case 'armedAway':
+        case 1:
+            return 'armAway';
+        case 'night':
+        case 'armnight':
+        case 'armNight':
+        case 'armednight':
+        case 'armedNight':
+        case 2:
+            return 'armNight';
+        case 'off':
+        case 'disarm':
+        case 'disarmed':
+        case 'allDisarmed':
+        case 'alldisarmed':
+        case 3:
+            return 'disarm';
+      }
+      return 'invalid';      
+    }
+
+    node.on('input', async (msg, send, done) => {
+
+      let { command } = node;
+      if (msg.command !== undefined)
+        command = msg.command;
+      if (!command) {
+        if (node.currentHsm === undefined) {
+          node.status({ fill: 'red', shape: 'ring', text: 'unitialized' });
+          done('unitialized');
+        } else {
+          const output = {
+            ...msg,
+            payload: { name: 'hsmStatus', value: node.currentHsm },
+          };
+          send(output);
+          node.status({ fill: 'blue', shape: node.shape, text: node.currentHsm });
+          done();
+        }
+        return;
+      }
+      node.status({ fill: 'blue', shape: node.shape, text: 'requesting' });
+      command = convertAlarmState(command);
+
+      if (command === 'invalid') {
+        node.status({ fill: 'red', shape: 'ring', text: 'invalid command' });
+        done('invalid command');
+        return;
+      }
+
+      const url = `${node.hubitat.baseUrl}/hsm/${command}?access_token=${node.hubitat.token}`;
+      const options = { method: 'GET' };
+
+      try {
+        const response = await fetch(url, options);
+        if (response.status >= 400) {
+          node.status({ fill: 'red', shape: 'ring', text: 'response error' });
+          done(await response.text());
+          return;
+        }
+        node.currentHsm = output.response.hsm;
```

Comment:
I don't understand how it supposed to work, since output is not defined.
Is all node functionalities have been tested?

Reply:
That looks like I made a mistake here. That was supposed to be "response.hsm" as output get's defined after that.